sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/ExactMatcher.java

```diff
@@ -20,15 +21,28 @@ public static StringMatcher createAttribute(String name, String value) {
         return create(AttributeStringUtils.join(name, value));
     }
 
-    private final String str;
+    private String str;
+    private static final Double epsilon = 0.0001;
+    private static final DecimalFormat format = new DecimalFormat("0.###");
 
     @JsonCreator
     private ExactMatcher(@JsonProperty("string") String str) {
-        this.str = str;
+        try {
+            this.str = format.format(Double.valueOf(str));
+        } catch (NumberFormatException ex) {
+            this.str = str;
+        }
     }
 
     @Override
     public boolean matches(String value) {
+        try {
+            Double in = Double.valueOf(value);
+            value = format.format(in);
+        } catch (NumberFormatException ex) {
+            // Continue with original value of `value`
```

Comment:
Wouldn't this let you set a bogus value and then keep it?

Reply:
Nevermind, read further. Only question is what use cases involve double-valued attributes, it scares me a little given floating point behavior.

Reply:
I don't know of any use case.  It's just part of the spec.  I'll send you a link elsewhere.